source/modulo_core/modulo_core/translators/message_writers.py

```diff
@@ -14,6 +14,46 @@
 StateT = TypeVar('StateT')
 
 
+def get_clproto_msg_type(state: StateT) -> clproto.MessageType:
+    if isinstance(state, sr.CartesianPose):
```

Comment:
minor nitpick: the `isinstance(state, ...)` checks are useful and easily readable for this case, but there are two small downsides. One is that you have to check each type until it passes, so you're calling it many times. The other thing is that it means you have to be intentionally checking in the order from most derived to least derived, because `isinstance` returns true for subclasses (`isinstance(sr.CartesianPose(), sr.CartesianState)` would be True). That could be brittle to maintain and introduce hard to find bugs if we mix it up in future.

it might be better to do   better something like `state.get_type()` (after verifying that it has the corresponding callable attribute) like:

```python3
if not hasattr(state, 'get_type') or not callable(state.get_type):
    return clproto.MessageType.UNKNOWN_MESSAGE

type = state.get_type()
if type == sr.StateType.CARTESIAN_STATE:
   ...
```

Reply:
Good point thanks!